diff -urN boost-mpi3-v0.81-orig/CMakeLists.txt boost-mpi3-v0.81-patched/CMakeLists.txt
--- boost-mpi3-v0.81-orig/CMakeLists.txt	2023-02-14 02:01:06.000000000 +0100
+++ boost-mpi3-v0.81-patched/CMakeLists.txt	2025-03-04 13:31:10.937338461 +0100
@@ -2,29 +2,57 @@
 cmake_minimum_required(VERSION 3.16)
 
 project(
-	mpi3
-	VERSION 0.79.0
-	DESCRIPTION "B-MPI3 is a C++ library wrapper for version 3.1 of the MPI standard interface that simplifies the utilization and maintenance of MPI code."
-	HOMEPAGE_URL "https://gitlab.com/correaa/boost-mpi3"
-	LANGUAGES CXX
-)
+  bmpi3
+  VERSION 0.81.0
+  DESCRIPTION
+    "B-MPI3 is a C++ library wrapper for version 3.1 of the MPI standard interface that simplifies the utilization and maintenance of MPI code."
+  HOMEPAGE_URL "https://gitlab.com/correaa/boost-mpi3"
+  LANGUAGES CXX)
 
 include(GNUInstallDirs)
 
-add_library(${PROJECT_NAME} INTERFACE)
-
-target_include_directories(${PROJECT_NAME} INTERFACE $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}> $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include/mpi3/dummy>)
+# Find the dependencies.
+find_package(MPI REQUIRED COMPONENTS CXX) # C)  # might need to `module load
+                                          # mpi`
+find_package(Boost REQUIRED COMPONENTS serialization)
 
+# Set up the (interface) library.
+add_library(${PROJECT_NAME} INTERFACE)
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_17)
-#set_target_properties(${PROJECT_NAME} PROPERTIES CXX_EXTENSIONS OFF)
-
-# this makes CM FetchContent friendly https://www.foonathan.net/2022/06/cmake-fetchcontent/
-if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-  	return()
+target_include_directories(
+  ${PROJECT_NAME}
+  INTERFACE $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include/mpi3/dummy>)
+target_link_libraries(${PROJECT_NAME} INTERFACE MPI::MPI_CXX)
+target_link_libraries(${PROJECT_NAME} INTERFACE Boost::serialization)
+
+# Set up the installation of the library/project.
+option(BMPI3_INSTALL "Install the bmpi3 library with the current project" ON)
+if(BMPI3_INSTALL)
+  include(CMakePackageConfigHelpers)
+  install(
+    TARGETS ${PROJECT_NAME}
+    EXPORT ${PROJECT_NAME}Targets)
+  install(
+    DIRECTORY include/
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+  install(
+    EXPORT ${PROJECT_NAME}Targets
+    FILE "${PROJECT_NAME}-config-targets.cmake"
+    NAMESPACE "${PROJECT_NAME}::"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+  write_basic_package_version_file( "${PROJECT_NAME}-config-version.cmake"
+    VERSION "${PROJECT_VERSION}"
+    COMPATIBILITY AnyNewerVersion)
+  configure_package_config_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/cmake/${PROJECT_NAME}-config.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
+    PATH_VARS CMAKE_INSTALL_INCLUDEDIR
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+  install(
+    FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
+          "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 endif()
-
-include(CTest)
-
-enable_testing()
-
-add_subdirectory(test)
diff -urN boost-mpi3-v0.81-orig/cmake/bmpi3-config.cmake.in boost-mpi3-v0.81-patched/cmake/bmpi3-config.cmake.in
--- boost-mpi3-v0.81-orig/cmake/bmpi3-config.cmake.in	1970-01-01 01:00:00.000000000 +0100
+++ boost-mpi3-v0.81-patched/cmake/bmpi3-config.cmake.in	2025-03-04 13:21:50.287433068 +0100
@@ -0,0 +1,22 @@
+
+# Set up the helper functions/macros.
+@PACKAGE_INIT@
+
+# Set up some simple variables for using the package.
+set(BMPI3_VERSION "@PROJECT_VERSION@")
+set_and_check(BMPI3_INCLUDE_DIR "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@")
+set(BMPI3_INCLUDE_DIRS "${BMPI3_INCLUDE_DIR}")
+
+# Find all package dependencies.
+include(CMakeFindDependencyMacro)
+find_dependency(MPI COMPONENTS CXX)
+find_dependency(Boost COMPONENTS serialization)
+
+# Include the file listing all the imported targets and options.
+include("${CMAKE_CURRENT_LIST_DIR}/bmpi3-config-targets.cmake")
+
+# Print a standard information message about the package being found.
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(bmpi3
+  REQUIRED_VARS CMAKE_CURRENT_LIST_FILE
+  VERSION_VAR BMPI3_VERSION)
